### PR TITLE
Task/keycloak gold upgrade dev/2204

### PR DESCRIPTION
**Pull request by conbrad**
Migrates keycloak to the gold offering.

Locally, you will need to update your:
- `web/.env`
    - REACT_APP_KEYCLOAK_AUTH_URL
    - REACT_APP_KEYCLOAK_REALM
    - REACT_APP_KEYCLOAK_CLIENT
- `api/app/.env`
    - REACT_APP_KEYCLOAK_AUTH_URL
    - REACT_APP_KEYCLOAK_REALM
    - REACT_APP_KEYCLOAK_CLIENT
    - KEYCLOAK_PUBLIC_KEY
    - Add KEYCLOAK_CLIENT
# Test Links:
[Percentile Calculator](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2277.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: web/src/features/auth/slices/authenticationSlice.ts [hunk 1]
```diff
@@ -23,6 +24,7 @@ export const initialState: State = {
   isAuthenticated: false,
   tokenRefreshed: false,
   token: undefined,
+  idToken: undefined,
```
Comment (andrea-williams) [inline web/src/features/auth/slices/authenticationSlice.ts:27]: is `token` for Keycloak in silver and `idToken` for gold Keycloak?

File: web/src/features/auth/slices/authenticationSlice.ts [hunk 2]
```diff
@@ -160,15 +171,17 @@ export const authenticate = (): AppThunk => dispatch => {
...
+  async dispatch => {
+    try {
+      const postLogoutRedirectURI = window.location.href
+      const returl = encodeURIComponent(
+        `${KC_AUTH_URL}/realms/${KC_REALM}/protocol/openid-connect/logout?client_id=${KC_CLIENT}&id_token_hint=${idToken}&post_logout_redirect_uri=${postLogoutRedirectURI}`
+      )
+      const logoutURL = `${SM_LOGOUT_URL}${returl}`
+      window.location.href = logoutURL
+    } catch (e) {
+      return dispatch(signoutError(`Failed to signout: ${e}`))
```
Comment (andrea-williams) [inline web/src/features/auth/slices/authenticationSlice.ts:185]:
```suggestion
      return dispatch(signoutError(`Failed to sign out: ${e}`))
```

File: web/src/features/auth/slices/authenticationSlice.ts [hunk 3]
```diff
@@ -160,15 +171,17 @@ export const authenticate = (): AppThunk => dispatch => {
...
-    const logoutURL = `${SM_LOGOUT_URL}${returl}`
-    window.location.href = logoutURL
-  } catch (e) {
-    return dispatch(signoutError(`Failed to signout: ${e}`))
+export const signout =
+  (idToken?: string): AppThunk =>
+  async dispatch => {
+    try {
+      const postLogoutRedirectURI = window.location.href
+      const returl = encodeURIComponent(
```
Comment (andrea-williams) [inline web/src/features/auth/slices/authenticationSlice.ts:179]: I know I'm being v picky now, but I'd prefer a different name for this variable, so it's clearer what it is.

**Review by andrea-williams — APPROVED** [review]

File: web/src/features/auth/slices/authenticationSlice.ts (hunk 1 above)
Comment (conbrad) [reply web/src/features/auth/slices/authenticationSlice.ts:27]: Both are for gold keycloak. I believe the`idToken` existed in silver, but required as a parameter in the sign out URL for gold now.

**Review by Sybrand — APPROVED** [review]

**Pull request merged**